### ISSUE TITLE
[Suggestion] Abbreviated event titles

### ISSUE THREAD

**Issue by denydias** [sportstimes/f1#917]
It would be very nice to have abbreviated event titles, e.g.:

- **Now:** Free Practice 1 (Australian Grand Prix) (40 chars long string)
- **Suggested:** FP1 Australian GP (18 chars string, 55% shorter)

That way the calendar would not cobbler small devices screen and be way more readable than how it is now.

**Comment by georgefeakes** [comment]
I agree with this. Would be nice to have event details shorter so they fit nicer in the iPhone calendar widget.
Would also be nice to include the Formula division in front of the event for users that subscribe to multiple calendars. It can be tricky to know which event is for which division at first glance.

Format:
{Division} {Session} ({Location})

**Comment by ay8s** [comment]
Thanks @denydias @georgefeakes.

I'll be deploying a change shortly which will add the division to the title and move the GP name to the description.

I'll see about abbreviating the session name, localization makes this a little more challenging now.

**Comment by denydias** [comment]
Tks for the heads up, @ay8s. Translation may be difficult for start, but once it's done, it's done. See some F1 examples translated into Brazilian Portuguese:

- **FP:** TL (Free Practice/Treino Livre)
- **SR:** CS (Sprint Race/Corrida Sprint)
- **Quali:** Clas. (Qualifying/Classificação)
- **GP:** GP (Grand Prix/Grande Prêmio)

Of course there are idioms out there where this could be trouble, but the challenge might be interesting.

**Comment by ay8s** [comment]
@denydias @georgefeakes I've deployed a tweak to the calendars so the format is...
F1: Session (Location)

In the localization files if there is a scheduleAbbreviated value it'll use that. So right now we'll use FP1, FP2 and FP3 for English calendars.

@denydias Feel free to add scheduleAbbreviated to pt-BR. Similar to this... https://github.com/sportstimes/f1/blob/main/locales/en/localization.json#L38-L62

I ended up tweaking yesterdays deploy after a bit of feedback from folks on Reddit and Twitter. Unfortunately it's tricky to please everyone so hoping shortening the lengthy ones like Free Practice 1 goes someway to smooth things out. 

I think one project I'd love to work on sometime in the future is improving the localization file so we can remove the duplication of "Grand Prix" from each string and have that just applied to the start or end depending on the locale. That way it'd be easier to use the long form or just "GP".

**Comment by denydias** [comment]
Sure, @ay8s! As there's already a pt-BR json, I'll just drop it bellow so you can add the `scheduleAbbreviated` object to that translation:

```json
"scheduleAbbreviated": {
    "fp1": "TL1",
    "fp2": "TL2",
    "fp3": "TL3",
    "fp4": "TL4",
    "shakedown": "Shakedown",
    "practice1": "Treino 1",
    "practice2": "Treino 2",
    "practice3": "Treino 3",
    "practice": "Treino",
    "qualifying": "Classificação",
    "qualifying1": "Classificação 1",
    "qualifying2": "Classificação 2",
    "sprint": "Sprint",
    "semifinal": "Semifinal",
    "warmup": "Aquecimento",
    "race": "Corrida",
    "race1": "Corrida 1",
    "race2": "Corrida 2",
    "race3": "Corrida 3",
    "gp": "Grande Prêmio",
    "sprint1": "Sprint 1",
    "sprint2": "Sprint 2",
    "feature": "Destaque"
}
```

PS1: there are no really good words in Brazilian Portuguese for `shakedown` and `sprint`. The closest ones are respectivelly `teste` and `rápida`, but they sound weird. Brazilian fans are used to these English terms in racing context (including `Q1`, `Q2` and `Q3` when using the abbreviatted word for qualifying (classificação)).

PS2: depending on how much you are willing to abbreviate, you might also use the ones bellow:

```json
"scheduleAbbreviated": {
    ...
    "practice1": "T1",
    "practice2": "T2",
    "practice3": "T3",
    "practice": "T",
    "qualifying": "Q",
    "qualifying1": "Q1",
    "qualifying2": "Q2",
    "sprint": "CS",
    ...
    "race": "C",
    "race1": "C1",
    "race2": "C2",
    "race3": "C3",
    "gp": "GP",
    "sprint1": "CS1",
    "sprint2": "CS2",
    ...
}
```

PS3: is `qualifying3` (e.g. Q3) missing?

PS4: yeah... it's just impossible to please everyone. My personal method: don't ask, just do what I want/need/feel to.

**Comment by ay8s** [comment]
Thanks @denydias.

I'll get that added to pt-BR. I'll rely on anyone familiar with the language to pick the best abbreviations to use. Leaning on using the same terms broadcasters etc user.

As for Q3, we use qualifying for F1 and don't break down the individual sessions. qualifying1 and qualifying2 relate to other series that have multiple qualifying sessions during a race weekend.

Once I've got pt-BR updated I'll close this out as I think we've got it covered and other languages can adopt it over time.
